buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
+							t.Fatalf("Unexpected error:\n%s\n", err)
+						}
 
-			it("should ensure the buildpack's layers dir exists and process build layers", func() {
-				h.Mkdir(t,
-					filepath.Join(layersDir, "A"),
-					filepath.Join(appDir, "layers-A-v1", "layer1"),
-					filepath.Join(appDir, "layers-A-v1", "layer2"),
-					filepath.Join(appDir, "layers-A-v1", "layer3"),
-				)
-				h.Mkfile(t, "[types]\n  build = true",
-					filepath.Join(appDir, "layers-A-v1", "layer1.toml"),
-					filepath.Join(appDir, "layers-A-v1", "layer3.toml"),
-				)
-				// the testdata/buildpack/bin/build script copies the content of the appDir into the layersDir
-				gomock.InOrder(
-					mockEnv.EXPECT().AddRootDir(filepath.Join(layersDir, "A", "layer1")),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer1", "env"), env.ActionTypeOverride),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer1", "env.build"), env.ActionTypeOverride),
-				)
-				gomock.InOrder(
-					mockEnv.EXPECT().AddRootDir(filepath.Join(layersDir, "A", "layer3")),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer3", "env"), env.ActionTypeOverride),
-					mockEnv.EXPECT().AddEnvDir(filepath.Join(layersDir, "A", "layer3", "env.build"), env.ActionTypeOverride),
-				)
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				testExists(t,
-					filepath.Join(layersDir, "A"),
-				)
-			})
+						var actual string
+						t.Log("sets CNB_BUILDPACK_DIR")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-buildpack-dir-A-v1.clear"))
+						h.AssertEq(t, actual, descriptor.Dir)
 
-			it("should provide the platform dir", func() {
-				h.Mkfile(t, "some-data",
-					filepath.Join(platformDir, "env", "SOME_VAR"),
-				)
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				testExists(t,
-					filepath.Join(appDir, "build-env-A-v1", "SOME_VAR"),
-				)
-			})
+						t.Log("sets CNB_PLATFORM_DIR")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-platform-dir-A-v1.clear"))
+						h.AssertEq(t, actual, platformDir)
 
-			it("should provide environment variables", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
-					t.Fatalf("Unexpected error:\n%s\n", err)
-				}
-				if s := cmp.Diff(h.Rdfile(t, filepath.Join(appDir, "build-info-A-v1")),
-					"TEST_ENV: Av1\n",
-				); s != "" {
-					t.Fatalf("Unexpected info:\n%s\n", s)
-				}
-			})
+						t.Log("sets CNB_BP_PLAN_PATH")
+						actual = h.Rdfile(t, filepath.Join(appDir, "build-env-cnb-bp-plan-path-A-v1.clear"))
+						if isUnset(actual) {
+							t.Fatal("Expected CNB_BP_PLAN_PATH to be set")
+						}
 
-			it("should set CNB_BUILDPACK_DIR", func() {
-				if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err != nil {
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L187-L222